frontend/src/pages/signin-page.tsx

```diff
@@ -0,0 +1,109 @@
+import { Link, useNavigate } from 'react-router-dom';
+import AddGoogleIcon from '@/assets/svg/google-color-icon.svg';
+import AddGithubIcon from '@/assets/svg/github-icon.svg';
+import { useForm } from 'react-hook-form';
+import type { FieldValues } from 'react-hook-form';
+import { TSignInSchema, signInSchema } from '@/lib/types';
+import 'react-toastify/dist/ReactToastify.css';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { toast } from 'react-toastify';
+
+function signin() {
+  const navigate = useNavigate();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+    reset,
+  } = useForm<TSignInSchema>({ resolver: zodResolver(signInSchema) });
+
+  const onSubmit = async (data: FieldValues) => {
+    if (data.email === 'abc@gamil.com') {
+      toast.error('Submitting form is failed');
+      return;
+    }
+
+    // TODO: Server-side validation
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+
+    reset();
+    navigate('/');
+  };
+
+  return (
+    <div className="m-4 flex-grow cursor-default bg-white py-4">
+      <div className="mb-4 flex justify-center">
+        <div className="flex w-full items-center justify-center">
+          <h2 className="w-3/4 text-center text-lg font-bold text-black sm:text-xl">
+            Sign in to WanderLust
+          </h2>
+        </div>
+      </div>
+      <div className="m-2 mt-8 flex flex-col items-center justify-center gap-2">
+        <form onSubmit={handleSubmit(onSubmit)} className="w-full md:w-3/4 lg:w-2/5">
+          <div className="mb-2">
+            <input
+              {...register('email')}
+              type="email"
+              placeholder="Email"
+              className="w-full rounded-lg bg-zinc-100 p-3 font-normal placeholder:text-sm placeholder:text-neutral-500"
+            />
+            {errors.email && (
+              <p className="p-3 text-xs text-red-500">{`${errors.email.message}`}</p>
+            )}
+          </div>
+
+          <div className="mb-4">
```

Comment:
This has a same pattern of classnames for both signin and signup pages. Can we extract a component out and reuse in both the pages?